### PR TITLE
adding typing loss

### PR DESCRIPTION
**Pull request by autoih**
typing loss: focal_loss, lifted, npairs, triplet

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Thanks for the pull request! Can you fix the conflicts?

**Comment by autoih** [comment]
Thanks @gabrieldemarmiesse. One question, will we use quote, i.e., apostrophe for reduction: `str`? It looks both ways can pass the testing, [example](https://github.com/tensorflow/addons/pull/1007/files).

File: tensorflow_addons/losses/focal_loss.py [hunk 1]
```diff
@@ -64,12 +64,13 @@ class SigmoidFocalCrossEntropy(tf.keras.losses.Loss):
...
           `gamma` is less than zero
     """
+
     @typechecked
     def __init__(self,
                  from_logits: bool = False,
                  alpha: Number = 0.25,
                  gamma: Number = 2.0,
-                 reduction: str = tf.keras.losses.Reduction.NONE,
+                 reduction: str = 'tf.keras.losses.Reduction.NONE',
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/focal_loss.py]:
```suggestion
                 reduction: str = tf.keras.losses.Reduction.NONE,
```

```
>>> print(type(tf.keras.losses.Reduction.NONE))
str
```

File: tensorflow_addons/losses/lifted.py [hunk 2]
```diff
@@ -17,10 +17,16 @@
...
+from tensorflow_addons.utils.types import FloatTensorLike, Number, TensorLike
+from typeguard import typechecked
+from typing import Optional
+
 
 @tf.keras.utils.register_keras_serializable(package="Addons")
 @tf.function
-def lifted_struct_loss(labels, embeddings, margin=1.0):
+def lifted_struct_loss(
+    labels: FloatTensorLike, embeddings: TensorLike, margin: Number = 1.0
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/lifted.py]:
```suggestion
    labels: TensorLike, embeddings: TensorLike, margin: Number = 1.0
```

I really need to change the name to ScalarTensor.

File: tensorflow_addons/losses/lifted.py [hunk 3]
```diff
@@ -17,10 +17,16 @@
...
+) -> FloatTensorLike:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/lifted.py]:
```suggestion
) -> tf.Tensor:
```

File: tensorflow_addons/losses/npairs.py [hunk 4]
```diff
@@ -16,10 +16,13 @@
...
 import tensorflow as tf
 
+from tensorflow_addons.utils.types import TensorLike, FloatTensorLike
+from typeguard import typechecked
+
 
 @tf.keras.utils.register_keras_serializable(package="Addons")
 @tf.function
-def npairs_loss(y_true, y_pred):
+def npairs_loss(y_true: TensorLike, y_pred: TensorLike) -> FloatTensorLike:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/npairs.py]:
```suggestion
def npairs_loss(y_true: TensorLike, y_pred: TensorLike) -> tf.Tensor:
```

File: tensorflow_addons/losses/npairs.py [hunk 5]
```diff
@@ -61,7 +64,7 @@ def npairs_loss(y_true, y_pred):
 
 @tf.keras.utils.register_keras_serializable(package="Addons")
 @tf.function
-def npairs_multilabel_loss(y_true, y_pred):
+def npairs_multilabel_loss(y_true: TensorLike, y_pred: TensorLike) -> FloatTensorLike:
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/losses/npairs.py]:
```suggestion
def npairs_multilabel_loss(y_true: TensorLike, y_pred: TensorLike) -> tf.Tensor:
```

File: tensorflow_addons/losses/focal_loss.py (hunk 1 above)
Comment (autoih) [reply tensorflow_addons/losses/focal_loss.py]: Thanks @gabrieldemarmiesse.
